src/routes/root/Summary.tsx

```diff
@@ -1,73 +1,147 @@
 import { useNavigate } from 'react-router-dom';
 import Contract from 'web3-eth-contract';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Heading } from '../../components/Heading';
 import { Button } from '../../components/Button';
 import { useEnsuredWallet } from '../../context/WalletContext';
-import { DemoAbi as Abi } from '../../abi';
-import { getNetwork, Networks } from '../../utils/network';
+import { Networks, getNetwork } from '../../utils/network';
+import { DepositAdapterABI, DEPOSIT_ADAPTER_ORIGIN } from '../../contracts';
+import { DepositDataJSON } from '../../components/JSONDropzone/validation';
+import { useStorage } from '../../context/StorageContext';
+
+// mocks
+const value = '{value}';
+const currency = '{currency}';
 
 export function Summary(): JSX.Element {
   const wallet = useEnsuredWallet();
-  const [network, setNetwork] = useState<Networks | null>(null);
+  const storage = useStorage();
+  const navigate = useNavigate();
+
+  const [selectedNetwork, setSelectedNetwork] = useState<Networks | null>(null);
+  const [manuallyChaingedNetwork, setManuallyChaingedNetwork] = useState<string | null>(
+    null,
+  );
+
   const [errorMsg, setErrorMsg] = useState<string>();
 
-  // mocks
-  const value = '{value}';
-  const currency = '{currency}';
+  useEffect(() => {
+    if (wallet.web3.currentProvider) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const provider = wallet.web3.currentProvider as any;
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+      provider.on('chainChanged', (chainId: number) => {
+        setManuallyChaingedNetwork(chainId.toString());
+      });
+    }
+  }, [wallet.web3]);
 
-  const navigate = useNavigate();
-  const handleBridgeClick = (): void => {
-    console.log('do magic on click');
+  //TODO - improve code
+  const handleBridgeClick = async (): Promise<void> => {
+    if (!selectedNetwork || !storage.data.json) return;
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const accounts = await wallet.web3.eth.getAccounts();
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     // @ts-ignore
```

Comment:
update web3 to 4.0.1-rc.0 and remove this. Note with that version, contract is export from web3 so you no longer need web3-eth-contract dep